pkg/prompts/prompts.go

```diff
@@ -629,3 +632,48 @@ func GetSubnetAuthKeys(prompt Prompter, controlKeys []string, threshold uint32)
 	}
 	return subnetAuthKeys, nil
 }
+
+func GetFujiKeyOrLedger(prompt Prompter, keyDir string) (bool, string, error) {
+	useStoredKey, err := prompt.ChooseKeyOrLedger()
+	if err != nil {
+		return false, "", err
+	}
+	if !useStoredKey {
+		return true, "", nil
+	}
+	keyName, err := captureKeyName(prompt, keyDir)
+	if err != nil {
+		if err == errNoKeys {
+			ux.Logger.PrintToUser("No private keys have been found. Deployment to fuji without a private key " +
+				"or ledger is not possible. Create a new one with `avalanche key create`, or use a ledger device.")
+		}
+		return false, "", err
+	}
+	return false, keyName, nil
+}
+
+func captureKeyName(prompt Prompter, keyDir string) (string, error) {
+	files, err := os.ReadDir(keyDir)
+	if err != nil {
+		return "", err
+	}
+
+	if len(files) < 1 {
+		return "", errNoKeys
+	}
+
+	keys := make([]string, len(files))
+
+	for i, f := range files {
+		if strings.HasSuffix(f.Name(), constants.KeySuffix) {
```

Comment:
I think you might want to use append instead of allocating a fixed length array. There could be some empty spots if there are some garbage files in the directory and this `if` doesn't trigger